backend/shared/cosmo_db.py

```diff
@@ -11,32 +11,122 @@
-        db = client.get_database_client(database=AZURE_DB_NAME)
-        container = db.get_container_client("reports")
         logging.info("Connection to Cosmos DB established successfully.")
         return container
     except Exception as e:
         logging.error(f"Error connecting to Cosmos DB: {e}")
         raise
 
-def create_reports(container, data):
+def create_report(data):
+    """
+    Creates a new document in the container.
+    """
     try:
+        credential = DefaultAzureCredential()
+        client = CosmosClient(AZURE_DB_URI, credential, consistency_level="Session")
+        db = client.get_database_client(database=AZURE_DB_NAME)
+        container = db.get_container_client("reports")
         data["id"] = str(uuid.uuid4())
+        data["generatedAt"] = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
         container.upsert_item(data)
-        logging.info(f"Document insert: {data}")
+        logging.info(f"Document created: {data}")
+        return data
     except Exception as e:
         logging.error(f"Error inserting data into Cosmos DB: {e}")
         raise
 
-def get_reports(container, query):
+def get_report(report_id):
+    """
+    Gets a specific document using its `id` as partition key.
+    """
+    credential = DefaultAzureCredential()
+    client = CosmosClient(AZURE_DB_URI, credential, consistency_level="Session")
+    db = client.get_database_client(database=AZURE_DB_NAME)
+    container = db.get_container_client("reports")
+
+    try:
+        report = container.read_item(
+            item=report_id, partition_key=report_id
+        )
+        logging.info(f"Report retrieved: {report}")
+        return report
+    except Exception as e:
+        logging.error(f"Error retrieving report with id {report_id}: {e}")
+        raise
+
+def get_report_by_type(report_type):
+    """
+    Gets a specific document using its `type` attribute.
+    """
+    credential = DefaultAzureCredential()
+    client = CosmosClient(AZURE_DB_URI, credential, consistency_level="Session")
+    db = client.get_database_client(database=AZURE_DB_NAME)
+    container = db.get_container_client("reports")
+
+    query = "SELECT * FROM c WHERE c.type = @type"
+    parameters = [{"name": "@type", "value": report_type}]
     
     try:
-        items = list(container.query_items(query=query, enable_cross_partition_query=True))
-        logging.info(f"Successful query: {len(items)} items found.")
+        items = list(container.query_items(
+            query=query,
+            parameters=parameters,
+            enable_cross_partition_query=True  # Necessary if you are not sure about the partition key
+        ))
+        logging.info(f"Reports retrieved: {items}")
         return items
     except Exception as e:
-        logging.error(f"Error querying Cosmos DB: {e}")
+        logging.error(f"Error retrieving reports with type {report_type}: {e}")
+        raise
```

Comment:
@ramirezmorac2  Consider creating custom exception classes for known issues (e.g., ReportNotFoundError)